addons/website/static/src/snippets/s_popup/000.js

```diff
@@ -116,5 +116,58 @@ const PopupWidget = publicWidget.Widget.extend({
 
 publicWidget.registry.popup = PopupWidget;
 
+// Extending the popup widget with cookiebar functionality.
+// This allows for refusing optional cookies for now and can be
+// extended to picking which cookies categories are accepted.
+publicWidget.registry.cookies_bar = PopupWidget.extend({
+    selector: '#website_cookies_bar',
+    cookieDurationDays: 365,
+    events: Object.assign({}, PopupWidget.prototype.events, {
+        'click #cookies-consent-essential': '_onConsentEssentialClick',
+        'click #cookies-consent-all': '_onConsentAllClick',
+    }),
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     * @param acceptOptional Whether optional cookies were accepted or not
+     */
+    _onAcceptClick(acceptOptional) {
+        const acceptedCookieTypes = JSON.parse(get_cookie('accepted_cookie_types') || '{}');
+        Object.assign(acceptedCookieTypes, {'required': true, 'optional': acceptOptional});
+        set_cookie('accepted_cookie_types',
+            JSON.stringify(acceptedCookieTypes),
+            this.cookieDurationDays * 24 * 60 * 60, 'required');
+        // Prevent the modal from reopening again.
+        this._onHideModal();
```

Comment:
We are trying to reduce the used cookies but here we are actually creating a second cookie for the cookie bar. We could regroup both.
Indeed the cookie bar is using the popup mechanism which is setting the popup id as cookie name to flag it as read & closed, in this case `website_cookies_bar` which is the ID of the cookie snippet

Can't we simply move the `_onHideModal()` call first in this method (so that it will actually set the cookie `website_cookies_bar`) and then update that cookie val to `{'required': true, 'optional': acceptOptional}` and just read this cookie in the code? (ofc with a comment next to the cookie template to mention that it's id is used as the name of the cookie so no one will change that id blindly)

Your opinion?

There could even be another way, have a new property (smth like that) set to `true` in the `PopupWidget`, and just make your `_onConsentAllClick` and `_onConsentEssentialClick` change that property to `{'required': true, 'optional': xx}`
Then in the hide_modal of `PopupWidget` simply set the cookie to this property.
That will solve the double cookie issue + remove the duplication of the set_cookie and day computation here